From 4ad5d102210bf87a91611d34b0db974198960155 Mon Sep 17 00:00:00 2001
From: Dan Nicholson <nicholson@endlessm.com>
Date: Tue, 14 Feb 2017 16:17:53 -0600
Subject: [PATCH] Mount /sys in buildroot

The sysfs mount at /sys is a standard part of Linux systems. While the
majority of builds don't need it, it's pretty reasonable that some will.
In particular, systemd's network test expects to create open a udev
device from /sys/class/net/lo.

https://phabricator.endlessm.com/T15164
---
 build            | 4 ++++
 init_buildsystem | 5 +++++
 2 files changed, 9 insertions(+)

diff --git a/build b/build
index 18db5f2..baedd30 100755
--- a/build
+++ b/build
@@ -309,6 +309,7 @@ cleanup_and_exit () {
     else
 	umount -n $BUILD_ROOT/proc/sys/fs/binfmt_misc 2> /dev/null || true
 	umount -n $BUILD_ROOT/proc 2>/dev/null || true
+	umount -n $BUILD_ROOT/sys 2>/dev/null || true
 	umount -n $BUILD_ROOT/dev/pts 2>/dev/null || true
 	umount -n $BUILD_ROOT/dev/shm 2>/dev/null || true
 	umount -n $BUILD_ROOT/run/shm 2>/dev/null || true
@@ -327,9 +328,11 @@ fail_exit()
 setup_virtual_filesystems()
 {
     mkdir -p $BUILD_ROOT/proc
+    mkdir -p $BUILD_ROOT/sys
     mkdir -p $BUILD_ROOT/dev/pts
     mkdir -p $BUILD_ROOT/dev/shm
     mount -n -tproc none $BUILD_ROOT/proc || true
+    mount -n -tsysfs none $BUILD_ROOT/sys
     mount -n -tdevpts -omode=0620,gid=5 none $BUILD_ROOT/dev/pts
     if ! test -f $BUILD_ROOT/dev/shm/.mounted ; then
       mount -n -ttmpfs none $BUILD_ROOT/dev/shm
@@ -1530,6 +1533,7 @@ for SPECFILE in "${SPECFILES[@]}" ; do
 	echo "shell='$shell'" >> $BUILD_ROOT/.build/build.data
 	umount -n $BUILD_ROOT/proc/sys/fs/binfmt_misc 2> /dev/null || true
 	umount -n $BUILD_ROOT/proc 2> /dev/null || true
+	umount -n $BUILD_ROOT/sys 2> /dev/null || true
 	umount -n $BUILD_ROOT/dev/pts 2> /dev/null || true
 	umount -n $BUILD_ROOT/dev/shm 2>/dev/null || true
 	umount -n $BUILD_ROOT/run/shm 2>/dev/null || true
diff --git a/init_buildsystem b/init_buildsystem
index 8c47f69..279f954 100755
--- a/init_buildsystem
+++ b/init_buildsystem
@@ -100,6 +100,7 @@ cleanup_and_exit()
 # XXX: use stat -f /dev/pts/ -c %T  to check whether it's mounted and not suppress errors then?
     umount -n $BUILD_ROOT/proc/sys/fs/binfmt_misc 2> /dev/null || true
     umount -n $BUILD_ROOT/proc 2> /dev/null || true
+    umount -n $BUILD_ROOT/sys 2> /dev/null || true
     umount -n $BUILD_ROOT/dev/pts 2> /dev/null || true
     umount -n $BUILD_ROOT/dev/shm 2> /dev/null || true
     umount -n $BUILD_ROOT/run/shm 2> /dev/null || true
@@ -110,9 +111,11 @@ cleanup_and_exit()
 setup_virtual_filesystems()
 {
     mkdir -p $BUILD_ROOT/proc
+    mkdir -p $BUILD_ROOT/sys
     mkdir -p $BUILD_ROOT/dev/pts
     mkdir -p $BUILD_ROOT/dev/shm
     mount -n -tproc none $BUILD_ROOT/proc || true
+    mount -n -tsysfs none $BUILD_ROOT/sys
     mount -n -tdevpts -omode=0620,gid=5 none $BUILD_ROOT/dev/pts
     if ! test -f $BUILD_ROOT/dev/shm/.mounted ; then
       mount -n -ttmpfs none $BUILD_ROOT/dev/shm
@@ -125,6 +128,7 @@ clean_build_root()
 	test -n "$BUILD_ROOT" && {
 	    umount -n $BUILD_ROOT/proc/sys/fs/binfmt_misc 2> /dev/null || true
 	    umount -n $BUILD_ROOT/proc 2> /dev/null || true
+	    umount -n $BUILD_ROOT/sys 2> /dev/null || true
 	    umount -n $BUILD_ROOT/dev/pts 2> /dev/null || true
 	    umount -n $BUILD_ROOT/dev/shm 2> /dev/null || true
 	    umount -n $BUILD_ROOT/run/shm 2> /dev/null || true
@@ -551,6 +555,7 @@ if test -e $BUILD_IS_RUNNING ; then
     echo To be sure, we will build it again completely...
     umount -n $BUILD_ROOT/proc/sys/fs/binfmt_misc 2> /dev/null || true
     umount -n $BUILD_ROOT/proc 2> /dev/null
+    umount -n $BUILD_ROOT/sys 2> /dev/null
     umount -n $BUILD_ROOT/dev/pts 2> /dev/null
     umount -n $BUILD_ROOT/dev/shm 2> /dev/null
     umount -n $BUILD_ROOT/run/shm 2> /dev/null
-- 
2.5.5

